File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.js
/**
 * @param {number} k
 * @return {number}
 */
var smallestRepunitDivByK = function(k) {
    if(k==1) return 1;
    let num=0;
    
    for(let l=1;l<=k;l++){
        num=(num*10+1) % k;
        if(num==0) return l;
    }

    return -1;
};

File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        if(k==1) return 1;
        int num=0;

        for(int l=1;l<=k;l++){
            num=(num*10 + 1) % k;
            if(num==0) return l;
        }

        return -1; //not possible
    }
};